Sources/PrimerSDK/Classes/User Interface/ApplePayPresentationManager.swift

```diff
@@ -77,20 +79,48 @@ class ApplePayPresentationManager: ApplePayPresenting, LogReporter {
         return request
     }
 
-    func shippingContactFields(applePayOptions: PrimerApplePayOptions?) -> Set<PKContactField> {
-        guard applePayOptions?.shippingOptions?.isCaptureShippingAddressEnabled == true else {
-            return []
-        }
+    func mapContactFieldsFromOptions(applePayOptions: PrimerApplePayOptions?) -> (mappedShippingContactFields: Set<PKContactField>, mappedBillingContactFields: Set<PKContactField>) {
 
-        var fields: Set<PKContactField> = [.postalAddress]
+        var requiredShippingContactFields = Set<PKContactField>()
+        var requiredBillingContactFields = Set<PKContactField>()
 
-        if let additionalFields = applePayOptions?.shippingOptions?.additionalShippingContactFields {
-            additionalFields.forEach {
-                fields.insert($0.toPKContact())
+        // Map required shipping contact fields
+        if let shippingContactFields = applePayOptions?.shippingOptions?.shippingContactFields, !shippingContactFields.isEmpty {
+            shippingContactFields.forEach {
+                requiredShippingContactFields.insert($0.toPKContact())
             }
         }
 
-        return fields
+        // Map required billing contact fields
+        if let billingContactFields = applePayOptions?.billingOptions?.requiredBillingContactFields {
+            billingContactFields.forEach {
+                requiredBillingContactFields.insert($0.toPKContact())
+            }
+        }
+
+        // Handle deprecated `isCaptureBillingAddressEnabled`
+        if requiredBillingContactFields.isEmpty, applePayOptions?.isCaptureBillingAddressEnabled == true {
+            requiredBillingContactFields.insert(.postalAddress)
+            requiredBillingContactFields.insert(.name)
```

Comment:
[nit]: as per their documentation, `.name` is useless in case `. postalAddress` is passed.

Reply:
In other words, should this line `requiredBillingContactFields.insert(.name)` be deleted? @euljr, this comment may be valid for WEB SDK as well.

Reply:
Yes! The `postalAddress` and `name` were there previously and I didn't think about changing back then.